climateconnect/climateconnect
# Feat: Follow Button for Organizatons

## Before merging

- Merge migrations
- Add new env variable values for the following variables: `ORGANIZATION_FOLLOWER_TEMPLATE_ID
ORGANIZATION_FOLLOWER_TEMPLATE_ID_DE
ORG_PUBLISHED_NEW_PROJECT_TEMPLATE_ID
ORG_PUBLISHED_NEW_PROJECT_TEMPLATE_ID_DE`

## Description
Closes #479 

Implementation of a follow button for the organization page:
- Added 2 new notifications:

1.  Notification for following an organization
2.  Notification for new project created by organization a user follows

- Added a clickable button to the organization page with some accompanying text
- Clicking the button will make the user follow the organization and the organization administrators will receive a notification on the bell icon that someone has followed their organization, as well as an email
- Users are able to see the follow count of the organization if the organization has at least 1 follower
- Users are able to see who is following the organization by clicking the follow count
- Clicking the notification brings the user to the organization page and opens the list of followers and removes the notification
- Followers of an organization will receive a notification on the bell icon if a project is shared under that organization, as well as an email
- Clicking the notification will bring the user to the project page and remove the notification from the bell icon
- These features are not available to users who aren't logged in and are notified that they need to sign in to access them
- When a following user of a project or organization signs out while on a project or organization page, the button will update to say "follow"
- A user can change in their settings page if they want to receive email notifications or not for the 2 new notifications


## Test plan

- Navigate to an organization page while signed in to see and use the button
- Navigate to the settings page to see the 2 new checkboxes for email notification preferences
- Follow an organization with an account then publish a project under that organization to receive a notification

## Before landing

1. PR has meaningful title
1. `yarn lint` passes (frontend)
1. `yarn format` passes (frontend)
1. `make format` passes (backend)
